test/test_script.py

```diff
@@ -1204,19 +1204,23 @@ def test_cmd_while(self):
 
     def test_cmd_map(self):
         context = Metadata()
-        context["foo"] = "First:A; Second:B; Third:C"
-        context["bar"] = ["First:A", "Second:B", "Third:C"]
         foo_output = "1=FIRST:A; SECOND:B; THIRD:C"
         loop_output = "1=FIRST:A; 2=SECOND:B; 3=THIRD:C"
         alternate_output = "1=FIRST:2=A; SECOND:3=B; THIRD:4=C"
         # Tests with context
+        context["foo"] = "First:A; Second:B; Third:C"
         self.assertScriptResultEquals("$map(%foo%,$upper(%_loop_count%=%_loop_value%))", foo_output, context)
+        context["bar"] = ["First:A", "Second:B", "Third:C"]
         self.assertScriptResultEquals("$map(%bar%,$upper(%_loop_count%=%_loop_value%))", loop_output, context)
         # Tests with static inputs
         self.assertScriptResultEquals("$map(First:A; Second:B; Third:C,$upper(%_loop_count%=%_loop_value%))", loop_output, context)
+        # Tests for removing empty elements
+        context["baz"] = ["First:A", "Second:B", "", "Third:C"]
+        self.assertScriptResultEquals("$map(%baz%,$upper(%_loop_count%=%_loop_value%))", loop_output, context)
```

Comment:
This assertion passes, but it does not test what - I believe - the reader thinks it does: the problem is that after `context["baz"] = ....` the empty element is already removed (`$lenmulti(%baz%)` is 3 instead of 4), so whether the map function discards empty elements or not is irrelevant to the outcome of the assertion. The assertion still passes if the remove-empty-elements code in `func_map` is removed.

Reply:
I think I see what you're saying.  Does this latest commit address your concern?